lib/Transforms/AnnotateSecretness/AnnotateSecretness.cpp

```diff
@@ -0,0 +1,60 @@
+#include "lib/Transforms/AnnotateSecretness/AnnotateSecretness.h"
+
+#include <utility>
+
+#include "lib/Analysis/SecretnessAnalysis/SecretnessAnalysis.h"
+#include "mlir/include/mlir/Analysis/DataFlow/ConstantPropagationAnalysis.h"  // from @llvm-project
+#include "mlir/include/mlir/Analysis/DataFlow/DeadCodeAnalysis.h"  // from @llvm-project
+#include "mlir/include/mlir/Analysis/DataFlowFramework.h"  // from @llvm-project
+#include "mlir/include/mlir/IR/MLIRContext.h"              // from @llvm-project
+#include "mlir/include/mlir/Transforms/Passes.h"           // from @llvm-project
+
+namespace mlir {
+namespace heir {
+
+#define GEN_PASS_DEF_ANNOTATESECRETNESS
+#include "lib/Transforms/AnnotateSecretness/AnnotateSecretness.h.inc"
+
+struct AnnotateSecretness : impl::AnnotateSecretnessBase<AnnotateSecretness> {
+  using AnnotateSecretnessBase::AnnotateSecretnessBase;
+
+  void runOnOperation() override {
+    MLIRContext *context = &getContext();
+
+    DataFlowSolver solver;
+    solver.load<dataflow::DeadCodeAnalysis>();
+    solver.load<dataflow::SparseConstantPropagation>();
+    solver.load<SecretnessAnalysis>();
+
+    auto result = solver.initializeAndRun(getOperation());
+
+    if (failed(result)) {
+      getOperation()->emitOpError() << "Failed to run the analysis.\n";
+      signalPassFailure();
+      return;
+    }
+
+    // Add an attribute to the operations to show determined secretness
+    OpBuilder builder(context);
+    getOperation()->walk([&](Operation *op) {
+      if (op->getNumResults() == 0) return;
+      auto *secretnessLattice =
+          solver.lookupState<SecretnessLattice>(op->getResult(0));
```

Comment:
This is a debugging pass, but it might be worth making this a bit more complete with respect to multi-result ops.

Maybe loop over op results and add the result index to the attribute name?

`result_{i}_secretness`

Or maybe only do this if there is more than one result.

Reply:
Ohh, good point - I'll add that tomorrow!

Reply:
Done :)